src/main/java/org/takes/facets/auth/social/PsTwitter.java

```diff
@@ -64,57 +71,112 @@
     private final transient String key;
 
     /**
+     * Request for fetching app trequest.
+     */
+    private final transient com.jcabi.http.Request trequest;
+
+    /**
+     * Request for verifying user credentials.
+     */
+    private final transient com.jcabi.http.Request user;
+
+    /**
      * Ctor.
-     * @param gapp Twitter app
-     * @param gkey Twitter key
+     * @param name Twitter app
+     * @param keys Twitter key
      */
-    public PsTwitter(final String gapp, final String gkey) {
-        this.app = gapp;
-        this.key = gkey;
+    public PsTwitter(final String name, final String keys) {
+        this(
+            new JdkRequest(
+                new Href("https://api.twitter.com/oauth2/trequest")
+                    .with("grant_type", "client_credentials")
+                    .toString()
+            ),
+            new JdkRequest(PsTwitter.VERIFY_URL),
+            name,
+            keys
+        );
+    }
+
+    /**
+     * Ctor with proper requestor for testing purposes.
+     * @param ttrequest HTTP request for getting trequest
+     * @param tvcrequest HTTP request for verifying credentials
+     * @param tapp Facebook app
+     * @param tkey Facebook key
+     * @checkstyle ParameterNumberCheck (3 lines)
```

Comment:
@adko-pl I don't think this is necessary. place the last 2 parameters on the same line.